src/coprocessor/codec/mysql/json/json_remove.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::Json;
+use super::path_expr::{PathLeg, PathExpression};
+use super::super::Result;
+
+impl Json {
+    // Remove elements from Json,
+    // All path expressions cannot contain * or ** wildcard.
+    // If any error occurs, the input won't be changed.
+    pub fn remove(&mut self, path_expr_list: &[PathExpression]) -> Result<()> {
+        if path_expr_list.iter().any(|expr| expr.legs.is_empty() || expr.contains_any_asterisk()) {
+            return Err(box_err!("Invalid path expression"));
+        }
+
+        for expr in path_expr_list {
+            self.remove_path(&expr.legs);
+        }
+        Ok(())
+    }
+
+    fn remove_path(&mut self, path_legs: &[PathLeg]) {
+        let (current_leg, sub_path_legs) = (&path_legs[0], &path_legs[1..]);
+
+        if let PathLeg::Index(index) = *current_leg {
+            if let Json::Array(ref mut array) = *self {
+                let index = index as usize;
+                if array.len() <= index {
+                    return;
```

Comment:
This logic was translated from `tidb` https://github.com/pingcap/tidb/blob/master/util/types/json/functions.go#L345, and it seems mysql would not return error in this case too .